docs/docs/reference/principled-meta-programming.md

```diff
@@ -72,13 +72,18 @@ The two types can be defined in package `scala.quoted` as follows:
 
     package scala.quoted
 
-    abstract class Expr[T] {
+    sealed abstract class Expr[T] {
       def unary_~: T   // splice operation
     }
-    class Type[T] {
+    sealed abstract class Type[T] {
```

Comment:
@odersky I created #3883 to make these classes `sealed`, no need to implement it on this PR.